packages/gatsby-theme-newrelic/src/components/GlobalHeader.js

```diff
@@ -31,13 +31,14 @@ const styles = {
 const GlobalHeader = ({ editUrl, className, search }) => {
   const location = useLocation();
   const [isOverlayOpen, setIsOverlayOpen] = useState(
-    Boolean(qs.parse(location.search).q)
+    Boolean(qs.parse(location.search).q?.length > 0)
```

Comment:
I'm sorry for soooo many comments. Perhaps I'm not seeing something obvious 😄 

Wouldn't this result in the same thing as before if `q` is set to an empty string? If my query string was `?q=`, that would mean the value of `q` is an empty string. The `q` _property_ is still present, so I want this to return `true`. 

If I'm understanding this properly, `qs.parse(location.search)` is going to result in an object that looks like the following: `{ q: '' }`. Following the above, this means that `qs.parse(location.search).q === ''` and `qs.parse(location.search).q.length === 0`. `Boolean(0)` is `false`, but we want `true` in this case because the `q` query parameter itself is present, regardless of its value. 

You might actually have an easier time with [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams?q=). It has a `has` method that lets you easily check for this.

```js
const query = new URLSearchParams(location.search)
query.has('q') // => true if `q` is a query param regardless of its value
```

The equivalent using `qs.parse` would be checking whether the key exists in the object:

```js
const query = qs.parse(location.search)
Object.prototype.hasOwnProperty.call(query, 'q')
```

Reply:
No you are so right, that was dumb on my part hahaha I changed it and pushed the work up!